docs/observation/README.md

```diff
@@ -0,0 +1,99 @@
+
+All repositories that need observation will have the **helix-bot** installed. The helix-bot doesn't know (and we don't store) the OpenWhisk credentials in the `helix-bot-config.yaml`. This is deliberate, to avoid distributing the credentials too broadly.
+
+the OpenWhisk credentials are needed to:
+- create a trigger that will receive the repository change events
+- fire the trigger when a change event happens
+
+To help with setting up observation, there is a `helix-services/observation` action, which implements a [feed](https://github.com/apache/openwhisk/blob/master/docs/feeds.md):
+- when creating the **trigger**, a random password is generated (`trigger-token`) and stored as annotation with the trigger.
+- the trigger is created using the `helix-services/observation` feed, which will:
+  - verify that the respective repository has a helix-bot setup
+  - update the `helix-bot-config` with the `trigger-namespace` and `trigger-token`
+- the `helix-services/observation` also forwards events received via `POST /feed/:owner/:repo/:ref` to the respective trigger.
+
+### Authorization
+
+Only events that either include `trigger-token` or are properly signed are forwarded to the trigger.
+- if the events contain a `x-trigger-token` header:
+  - the `helix-services/observation` action reads (and caches) the `trigger-token` annotation form the respective trigger
+  - if the trigger tokens don't match, it returns with a 403.
+
+- if the events contain a `x-bot-signature` header:
+  - the `helix-services/observation` action checks that the payload is signed with the bot's private key.
+  - if the signature is invalid, it returns a 403.
+
+**Note**:
+
+currently, the helix-bot is the only client that uses the `helix-services/observation` action directly, so we could
+drop the `trigger-token` feature. but for testing and potential other clients, it is still useful to have.
+
+### helix project (not helix-pages)
+
+#### bot setup
+similar to the way the purge configuration is updated during `hlx publish`, the step also:
+- ensures there's a trigger `{owner}--{repo}--{ref}` for all content repositories where a bot is installed
+- if not, create a trigger using the ``helix-services/observation` feed.
+
+#### additional content change listeners
+In order to register additional content change listeners, for example for onedrive, the setup is similar.
+
+#### onedrive
+for onedrive, each content repository that contains a `fstab.yaml` with mountpoints with a onedrive root need a _onedrive subscription_ that eventually invokes a change listener.
+- during `hlx publish` all `{repository, driveRoot}` tuples are used to update the subscriptions.
+- managing the subscriptions should probably happen in a separate action, but invoked by the `observation` feed.
+- each subscription needs to use the `onedrive-change-listener` action (webhook) in the respective customer namespace, so that the change listener can fire the trigger (without extra authentication).
```

Comment:
```suggestion
- each subscription needs to use the `onedrive-change-listener` action (webhook) in the respective customer namespace, so that the change listener can fire the trigger (without extra authentication). This action will be linked during `hlx deploy`.
```